addon/authorizers/devise-token-auth.js

```diff
@@ -0,0 +1,135 @@
+  uidHeaderAttributeName: 'uid',
+
+  /**
+    The token attribute name. __This will be used in the request and also be
+    expected in the server's response.__
+
+    @property tokenAttributeName
+    @type String
+    @default 'token'
+    @public
+  */
+  tokenHeaderTypeAttributeName: 'token-type',
+
+  /**
+    The identification attribute name.
+
+    @property identificationAttributeName
+    @type String
+    @default 'email'
+    @public
+  */
+  identificationAttributeName: 'email',
+
+  /**
+    Includes the appropriate headers (see
+    {{#crossLink "DeviseTokenAuthAuthenticator/tokenHeaderAttributeName:property"}}access-token{{/crossLink}},
+    {{#crossLink "DeviseTokenAuthAuthenticator/clientHeaderAttributeName:property"}}client{{/crossLink}},
+    {{#crossLink "DeviseTokenAuthAuthenticator/expiryHeaderAttributeName:property"}}expiry{{/crossLink}},
+    {{#crossLink "DeviseTokenAuthAuthenticator/uidHeaderAttributeName:property"}}uid{{/crossLink}},
+    and
+    {{#crossLink "DeviseTokenAuthAuthenticator/tokenHeaderTypeAttributeName:property"}}token-type{{/crossLink}}).
+
+    @method authorize
+    @param {Object} data The data that the session currently holds
+    @param {Function} block(headerName,headerContent) The callback to call with the authorization data; will receive the header name and header content as arguments
+    @public
+  */
+  authorize(data, block) {
+    const { tokenHeaderAttributeName,
+            clientHeaderAttributeName,
+            expiryHeaderAttributeName,
+            uidHeaderAttributeName,
+            tokenHeaderTypeAttributeName } = this.getProperties('tokenHeaderAttributeName',
+                                                                'clientHeaderAttributeName',
+                                                                'expiryHeaderAttributeName',
+                                                                'uidHeaderAttributeName',
+                                                                'tokenHeaderTypeAttributeName');
+    const tokenHeaderAttribute = get(data, tokenHeaderAttributeName);
+    const clientHeaderAttribute = get(data, clientHeaderAttributeName);
+    const expiryHeaderAttribute = get(data, expiryHeaderAttributeName);
+    const uidHeaderAttribute = get(data, uidHeaderAttributeName);
+    const tokenHeaderTypeAttribute = get(data, tokenHeaderTypeAttributeName);
+
+    if (!isEmpty(tokenHeaderAttribute)
+        && !isEmpty(clientHeaderAttribute)
+        && !isEmpty(expiryHeaderAttribute)
+        && !isEmpty(uidHeaderAttribute)
+        && !isEmpty(tokenHeaderTypeAttribute)
+        && parseInt(expiryHeaderAttribute) > Date.now()/1000) {
+      block(tokenHeaderAttributeName, tokenHeaderAttribute);
+      block(clientHeaderAttributeName, clientHeaderAttribute);
+      block(expiryHeaderAttributeName, expiryHeaderAttribute);
+      block(uidHeaderAttributeName, uidHeaderAttribute);
+      block(tokenHeaderTypeAttributeName, tokenHeaderTypeAttribute);
```

Comment:
The block can only be called once. Otherwise code like this:

```js
this.get('session').authorize('authorizer:devise-token-auth', () => {
  // make request
})
```

would result in 5 requests being sent.